rest_tools/server/handler.py

```diff
@@ -370,6 +373,8 @@ class OpenIDLoginHandler(RestHandler, OAuth2Mixin):
 
     Requires the `login_url` application setting to be a full url.
     """
+    _pkcs_challenges: MutableMapping[str, str] = TTLCache(maxsize=10000, ttl=3600)
```

Comment:
Any particular reason why you chose this instead of https://docs.python.org/3/library/functools.html#functools.cache?

Reply:
I wanted it to time out at some point (reject old challenges).  Here it's set to 1 hour.  Now, I don't think that's a guaranteed cutoff, since it only prunes when you touch the cache.  But it's a good general behavior, since the server will reject it at some point too.